drizzlepac/haputils/processing_utils.py

```diff
@@ -346,3 +346,54 @@ def make_section_str(str="", width=60, symbol='-'):
     side_dash = symbol * (dash_len // 2)
     section_str = "{}{}{}".format(side_dash, str, side_dash)
     return section_str
+
+def find_flt_keyword (hdu, keyword, extname="SCI", extver=1):
+    """Look for the FITS keyword in the Primary and specified extension header
+
+    This routine deliberately does not use astropy.io.fits getval() as it
+    should not be used in application code due to its self-documented inefficiency.
+
+    Parameters
+    ----------
+    hdu : HDUList
+    List of Header Data Unit objects
+
+    keyword : str
+    Keyword of interest
+
+    extname : str
+    Named extension to check in case the keyword is not found in the Primary
+    e.g., (extname, extver) ("SCI", 2)
+
+    extver : int
+    Specified extver of an extname to check in case the keyword is not found in the Primary
+    e.g., (extname, extver) ("SCI", 2)
+
+    Returns
+    -------
+    value : float
+    """
+
+    # Look for the keyword in the Primary header
+    try:
```

Comment:
Instead of an expensive `try...except` block, it may be easier to simply check for existence of the keyword in the headers before trying to get the value using 'if keyword in hdu[0].header`.  Only a suggestion as the current implementation would work just fine.

Reply:
Since this will be rewritten as a function more in line with what we will design, I leaving this function alone for now.